src/front/glsl/parser/declarations.rs

```diff
@@ -628,35 +628,55 @@ impl<'source> ParsingContext<'source> {
         loop {
             // TODO: type_qualifier
 
-            let (mut ty, mut meta) = self.parse_type_non_void(frontend)?;
-            let (name, end_meta) = self.expect_ident(frontend)?;
+            let (ty, mut meta) = self.parse_type_non_void(frontend)?;
 
-            meta.subsume(end_meta);
+            let (name, mut end_meta) = self.expect_ident(frontend)?;
+            let mut ty_copy = ty;
+            self.parse_array_specifier(frontend, &mut meta, &mut ty_copy)?;
 
-            self.parse_array_specifier(frontend, &mut meta, &mut ty)?;
+            let mut names_ty = vec![(name, ty_copy)];
```

Comment:
I think we can do away with this allocation if we immediately add to the members vector. My idea would be to do something like:

```rust
let (base_ty, mut meta) = self.parse_type_non_void(frontend)?;

loop {
	let (name, name_meta) = self.expect_ident(frontend)?;
    let mut ty = base_ty;
    self.parse_array_specifier(frontend, &mut meta, &mut ty_copy)?;

	meta.subsume(name_meta);

	let info = offset::calculate_offset(
        ty,
        meta,
        layout,
        &mut frontend.module.types,
        &mut frontend.errors,
    );

    let member_alignment = info.align;
    span = member_alignment.round_up(span);
    align = member_alignment.max(align);

    members.push(StructMember {
        name: Some(name),
        ty: info.ty,
        binding: None,
        offset: span,
    });

    span += info.span;

	if colon { continue; }

	self.expect(frontend, TokenValue::Semicolon)?;

	break;
}
```

This way we don't need to allocate the `names_ty` and don't need to duplicate code for the first name.

Reply:
Thanks @JCapucho for great review feedback! Changed the code in https://github.com/gfx-rs/naga/pull/2400/commits/85896caecfff94237e78602ba3358da05698f28a, could you check?